tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Process/ProcessStartCommon.cs

```diff
@@ -67,9 +72,213 @@ internal static Scope CreateScope(string filename, IDictionary<string, string> e
             return scope;
         }
 
-        internal static string Truncate(string value, int maxLength)
+        private static ProcessCommandStartTags PopulateTags(string filename, IDictionary<string, string> environmentVariables, bool useShellExecute, string arguments, Collection<string> argumentList)
+        {
+            // Environment variables
+            var variablesTruncated = EnvironmentVariablesScrubber.ScrubEnvironmentVariables(environmentVariables);
+            variablesTruncated = Truncate(variablesTruncated, MaxCommandLineLength, out _);
+
+            var tags = new ProcessCommandStartTags
+            {
+                EnvironmentVariables = variablesTruncated,
+                Component = "process",
+            };
+
+            // Don't populate further with command line information if shell collection is disabled
+            if (!Tracer.Instance.Settings.CommandsCollectionEnabled)
+            {
+                return tags;
+            }
+
+            if (useShellExecute)
+            {
+                // cmd.shell
+                var commandLine = filename;
+
+                // Append the arguments to the command line
+                if (!string.IsNullOrWhiteSpace(arguments))
+                {
+                    commandLine += " " + arguments;
+                }
+                else if (argumentList != null && argumentList.Count > 0)
+                {
+                    commandLine += " " + string.Join(" ", argumentList);
+                }
+
+                // Truncate the command line if needed
+                commandLine = Truncate(commandLine, MaxCommandLineLength, out var truncated);
+                tags.Truncated = truncated ? "true" : null;
+
+                tags.CommandShell = commandLine;
```

Comment:
you might use the `StringBuilderCache` here: 
```suggestion
                var commandLine = StringBuilderCache.Acquire(filename.Length);
                commandLine.Append(filename);
                // Append the arguments to the command line
                if (!string.IsNullOrWhiteSpace(arguments))
                {
                    commandLine.Append($" {arguments}");
                }
                else if (argumentList is { Count: > 0 })
                {
                    commandLine.Append($" {string.Join(" ", argumentList)}");
                }

                // Truncate the command line if needed
                var commandLineResult = Truncate(commandLine.ToString(), MaxCommandLineLength, out var truncated);
                tags.Truncated = truncated ? "true" : null;
                tags.CommandShell = commandLineResult;
```

Reply:
I did not know that one!